ci/run_envoy_docker.sh

```diff
@@ -78,7 +78,12 @@ if ! is_windows; then
     VOLUMES+=(-v "${SHARED_TMP_DIR}":"${SHARED_TMP_DIR}")
 fi
 
-time docker pull "${ENVOY_BUILD_IMAGE}"
+# Retry docker pull with 3 times
+i=0
+while [[ i -lt 3 ]]; do
+  time docker pull "${ENVOY_BUILD_IMAGE}" && break
```

Comment:
not sure this would work - its not catching the error

also i think we want some fail condition  if it fails after all retries

see https://github.com/envoyproxy/envoy/blob/14567bf0d16de6e052913a6582c369ac518c360e/examples/verify-common.sh#L149-L165 for a possible implementation that does both

Reply:
I test it with two image, one is existed another is not, both OK.

If you want to record more info, I will update it.

Reply:
hmm, doesnt make sense to me - its just not catching the error 

either way - i think we may be able to avoid adding this (for now) - see https://github.com/docker/hub-feedback/issues/2262#issuecomment-1197828295

Reply:
Covert to draft, let's see recent CI results.